rosidl_typesupport_c/resource/srv__type_support.cpp.em

```diff
@@ -105,10 +118,86 @@ static const type_support_map_t _@(service.namespaced_type.name)_service_typesup
   &_@(service.namespaced_type.name)_service_typesupport_data.data[0],
 };
 
+@#TODO(ihasdapie): Import Postfixes
+@{event_type = '__'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]) + SERVICE_EVENT_MESSAGE_SUFFIX}@
+@{request_type = '__'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]) + SERVICE_REQUEST_MESSAGE_SUFFIX}@
+@{response_type = '__'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]) + SERVICE_RESPONSE_MESSAGE_SUFFIX}@
+
+void *
+rosidl_typesupport_c_@('__'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]))__event_message__create(
```

Comment:
it sounds like we're rewriting the same function in many places, I think we need to move them to a common place

Reply:
Actually this is not that easy, as we need to create one function for each type.
Maybe we could generate the function once in `rosidl_generator_c/rosidl_generator_cpp` and use the correct one in each typesupport.

@wjwwood any opinion?

Reply:
I'm afraid I don't understand what you mean (I haven't read through these yet). We need a version of this for each service type and language (C/C++), which part are you hoping to de-duplicate?

Reply:
We're generating one per each typesupport.
One here, same thing for typesupport fastrtps (c and cpp), and same thing for typesupport introspection (c and cpp).

It should be one per language, not one per typesupport library.
I think that having the common implementation generated in `rosidl_generator_c`/`rosidl_generator_cpp` makes sense.
Then the typesupport implementation would only need to add that to the typesupport struct.

Reply:
See https://github.com/ros2/rosidl/pull/700/commits/0d6f816ce0a3f86566757294a419a2dcd7965ba4 (from https://github.com/ros2/rosidl/pull/700) and https://github.com/ros2/rosidl_typesupport/pull/127/commits/61d32cb9f1cfa5e8599017e468aee58bb7b6016c.
`rosidl_generator_c` was already generating the typesupport headers, so if there's a common function for C typesupports  it makes sense to generate it there.

I'm planning to do the same for the CPP typesupport soon.